docs: update the continuous integration story

CharlesSuttie opened the pull request:
This PR aims to update the CircleCi and Travis configurations in the continuous integration story.

The main changes are:

- update `ng test` and `ng e2e` command flags for version 6
- use `npm ci` instead of `npm install`
- use  headless chrome instead of chrome

To assist review:

- a link to the story in my fork is [here](https://github.com/CharlesSuttie/angular-cli/blob/docs-update-ci-story/docs/documentation/stories/continuous-integration.md).
-  A [demo Angular Cli app](https://github.com/CharlesSuttie/angular-cli-ci-story) has been created using the new instructions and configuration files proposed by this PR.

I have endeavored to maintain the style and simplicity of the story. All comments are very welcome.

@filipesilva, I noticed this [comment](https://github.com/angular/angular-cli/issues/11116#issuecomment-396180534) and would be happy to include what you said in this PR.

I feel this PR closes #10677 because the essential problem there was that the flags for the `ng test` and `ng e2e` commands changed with version 6. This PR also builds on #11445 and updates the `ng test` `--browsers flag` and the `ng e2e` `--protractor-config` flag, bringing the flags up to date with version 6.

CharlesSuttie commented:
I signed it!

filipesilva commented on docs/documentation/stories/continuous-integration.md:
```diff
@@ -76,28 +79,23 @@ jobs:
-
+      # install the latest npm to allow use of 'npm ci'
+      - run: sudo npm install -g npm@latest
+      - run: npm ci
```
Have you checked if `npm ci` is faster than using cache on the CI? Last time I checked `npm ci` was much slower.

CharlesSuttie replied:
Yes, you're quite right, using the cache is much faster.

filipesilva commented on docs/documentation/stories/continuous-integration.md:
```diff
@@ -23,22 +23,24 @@ set up Circle CI and Travis CI.
-  ChromeNoSandbox: {
-    base: 'Chrome',
-    flags: ['--no-sandbox']
+  myChromeHeadless: {
```
Nit: what do you think of `ChromeHeadlessCI` instead? I feel the `my` prefix doesn't say much about what's going on.

markgoho commented on docs/documentation/stories/continuous-integration.md:
```diff
@@ -76,28 +79,23 @@ jobs:
   build:
     working_directory: ~/my-project
     docker:
-      - image: circleci/node:8-browsers
```
When you provide only the major version, docker will always choose the most recent version.
E.g. `image: circleci/node:10-browsers` and then running `node -v` yields: `v10.6.0`

I would recommend reverting this change back to `- image: circleci/node:8-browsers`

CharlesSuttie replied:
Thanks @markgoho, good point!

CharlesSuttie commented:
Thanks @filipesilva ! I've rebased, squashed the commits into a single commit with an updated message (to remove the reference to npm ci). This is my first time contributing and I'm not totally confident I have the git part the way you would like, so please forgive me if I've made a mistake here.
